src/logging.c

```diff
@@ -78,31 +87,27 @@ rcutils_ret_t rcutils_logging_initialize_with_allocator(rcutils_allocator_t allo
       g_rcutils_logging_output_format_string[chars_to_copy] = '\0';
     } else {
       if (NULL != ret_str) {
-        fprintf(
-          stderr,
-          "Failed to get output format from env. variable: %s. Using default output format.\n",
-          ret_str);
+        RCUTILS_SET_ERROR_MSG(
+          "Failed to get output format from env. variable. Using default output format.",
+          g_rcutils_logging_allocator);
         ret = RCUTILS_RET_INVALID_ARGUMENT;
       }
       memcpy(g_rcutils_logging_output_format_string, g_rcutils_logging_default_output_format,
         strlen(g_rcutils_logging_default_output_format) + 1);
     }
 
-    if (!rcutils_allocator_is_valid(&allocator)) {
-      fprintf(
-        stderr,
-        "Provided allocator is invalid. Using the default allocator.\n");
-      ret = RCUTILS_RET_INVALID_ARGUMENT;
-      allocator = rcutils_get_default_allocator();
-    }
-    g_rcutils_logging_allocator = allocator;
     g_rcutils_logging_severities_map = rcutils_get_zero_initialized_string_map();
     rcutils_ret_t string_map_ret = rcutils_string_map_init(
       &g_rcutils_logging_severities_map, 0, g_rcutils_logging_allocator);
     if (string_map_ret != RCUTILS_RET_OK) {
-      fprintf(
-        stderr,
-        "Failed to initialize map for logger severities. Severities will not be configurable.\n");
+      // If an error message was set it will have been overwritten by rcutils_string_map_init.
+      char * msg = rcutils_format_string(
+        g_rcutils_logging_allocator,
+        "Failed to initialize map for logger severities [%s]. Severities will not be configurable.",
+        rcutils_get_error_string_safe());
+      rcutils_reset_error();
+      RCUTILS_SET_ERROR_MSG(msg, g_rcutils_logging_allocator)
```

Comment:
What if `msg` is `NULL`?

Reply:
Looks like this line could also set the error state: https://github.com/ros2/rcutils/blob/master/src/format_string.c#L41

Reply:
I've marked this PR as requiring https://github.com/ros2/rcutils/pull/63/ and swapped to use `RCUTILS_SET_ERROR_MSG_WITH_FORMAT_STRING` in https://github.com/ros2/rcutils/pull/65/commits/a088188e0ea0eb4d9bc0a57cb51342a5128fc20e which will take care of the NULL case.

I think I've missed your point regarding the format_string allocator check: I don't think that `RCUTILS_CHECK_ALLOCATOR` sets the error state or error message (`RCUTILS_CHECK_ALLOCATOR_WITH_MSG` [would though](https://github.com/ros2/rcutils/blob/00e2b9db9f4b71374f1388570be2de3edabd2e37/include/rcutils/allocator.h#L112))

Reply:
You're right, w.r.t. the check allocator call 👍